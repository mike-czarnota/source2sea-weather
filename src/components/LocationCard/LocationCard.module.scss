.self {
  transition-property: border-color, box-shadow;
}
