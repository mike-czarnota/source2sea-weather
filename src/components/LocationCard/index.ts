export { default } from './LocationCard';
export { default as LocationCardContainer } from './LocationCardContainer';
