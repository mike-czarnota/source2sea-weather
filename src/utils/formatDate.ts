import format from 'date-fns/format';

export const formatTime = (date: Date): string => format(date, 'HH:mm');
