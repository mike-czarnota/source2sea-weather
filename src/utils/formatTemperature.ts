export const formatTemperature = (temperature: number): string =>
  `${temperature}Â°C`;
