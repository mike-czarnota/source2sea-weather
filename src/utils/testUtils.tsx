import { render, RenderResult } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

export const renderWithWrapper = (ui: React.ReactElement): RenderResult =>
  render(<BrowserRouter>{ui}</BrowserRouter>);
