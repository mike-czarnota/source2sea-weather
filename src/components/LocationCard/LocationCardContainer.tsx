import { useQuery } from 'react-fetching-library';
import { IWeatherResponse } from '../../api/types/Weather';
import { getWeatherUrl } from '../../api/urls';
import Loader from '../Loader';
import { formatSimplifiedWeatherResponse } from '../../api/formatters/weather';
import LocationCard from './LocationCard';

interface ILocationCardContainerProps {
  locationName: string;
}

const LocationCardContainer: React.FC<ILocationCardContainerProps> = ({
  locationName,
}) => {
  const { loading, payload, error } = useQuery<IWeatherResponse>({
    method: 'GET',
    endpoint: getWeatherUrl(locationName.toLowerCase()),
  });

  if (loading) {
    return <Loader />;
  }

  if (error) {
    return (
      <div className="italic">
        Error while loading location:{' '}
        <span className="font-bold">{locationName}</span>
      </div>
    );
  }

  const data = formatSimplifiedWeatherResponse(payload as IWeatherResponse);

  return (
    <LocationCard name={data.name} temperatureInCelsius={data.currentTemp} />
  );
};

export default LocationCardContainer;
