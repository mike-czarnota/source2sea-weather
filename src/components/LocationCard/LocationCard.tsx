import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { formatTemperature } from '../../utils/formatTemperature';

import styles from './LocationCard.module.scss';
import { getLocationUrl } from '../../views/routes';

interface ILocationCardProps {
  name: string;
  temperatureInCelsius: number;
}

const LocationCard: React.FC<ILocationCardProps> = ({
  name,
  temperatureInCelsius,
}) => (
  <Link
    className={classNames(
      styles.self,
      `p-5 md:px-10 md:py-7 flex justify-between items-center border rounded-lg bg-white border-gray-300 hover:border-gray-900 duration-200 ease-in-out shadow-md hover:shadow-2xl`
    )}
    to={getLocationUrl(name)}
  >
    <span className="text-2xl">{name}</span>
    <span className="text-xl">{formatTemperature(temperatureInCelsius)}</span>
  </Link>
);

export default LocationCard;
