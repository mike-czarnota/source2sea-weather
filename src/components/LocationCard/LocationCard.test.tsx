import pretty from 'pretty';
import { screen } from '@testing-library/react';
import LocationCard from './LocationCard';
import { renderWithWrapper } from '../../utils/testUtils';

describe('testing LocationCard.test.tsx', () => {
  it('should render', () => {
    expect(
      pretty(
        renderWithWrapper(
          <LocationCard name="some random name" temperatureInCelsius={23} />
        ).baseElement.innerHTML
      )
    ).toMatchSnapshot();

    const locationName = screen.getByText('some random name');
    expect(locationName).toBeInTheDocument();

    const temperature = screen.getByText('23Â°C');
    expect(temperature).toBeInTheDocument();
  });
});
