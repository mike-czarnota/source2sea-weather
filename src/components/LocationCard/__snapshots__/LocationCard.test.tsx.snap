// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing LocationCard.test.tsx should render 1`] = `"<div><a class=\\"self p-5 flex justify-between items-center border rounded-lg bg-white border-gray-300 hover:border-gray-900 duration-200 ease-in-out shadow-md hover:shadow-2xl\\" href=\\"/some random name\\"><span class=\\"text-2xl\\">some random name</span><span class=\\"text-xl\\">23Â°C</span></a></div>"`;
